### PR TITLE
Fixes racecondition in tests

### PR DESCRIPTION
**Pull request by zeisss**
a fix for the race condition issue.

### CODE REVIEW

File: task/memory_storage.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 func NewMemoryStorage() Storage {
 	newStorage := &memoryStorage{
 		Mutex:   sync.Mutex{},
-		Storage: map[string]*TaskObject{},
```
Comment (zeisss) [inline task/memory_storage.go:12]: this ensures we are creating copies of our objects - so modifications to the object after saving does not change it anymore. same as a stored task in redis or etcd would be

File: task/task_service.go [hunk 2]
```diff
@@ -173,7 +173,7 @@ func (ts *taskService) WaitForFinalStatus(taskID string, closer <-chan struct{})
 		select {
 		case <-closer:
 			return nil, nil
-		default:
```
Comment (zeisss) [inline task/task_service.go:180]: this fixes the fact that `WaitForFinalStatus` always blocked 1s which lead to the race condition with in the Create test

File: task/task_service_test.go [hunk 3]
```diff
@@ -107,46 +107,37 @@ func Test_Task_TastService_Create_Wait(t *testing.T) {
...
-		fmt.Printf("%#v\n", 2)
-
+		// Just something to do, so the task blocks
 		time.Sleep(1 * time.Second)
-		fmt.Printf("%#v\n", 7)
 
 		return nil
 	}
 
-	fmt.Printf("%#v\n", 1)
```
Comment (zeisss) [inline task/task_service_test.go]: some renaming I did while playing around here. Not really needed, but might make it more readable

**Comment by zeisss** [comment]
ping @xh3b4sd

File: task/task_service.go [hunk 4]
```diff
@@ -173,7 +173,7 @@ func (ts *taskService) WaitForFinalStatus(taskID string, closer <-chan struct{})
...
+		case <-time.After(1 * time.Second):
```
Comment (xh3b4sd) [inline task/task_service.go:180]: Please merge the origin feature branch and use the configured `WaitSleep` here.

**Comment by xh3b4sd** [comment]
Please fix the comment. LGTM

**Pull request merged**